app/javascript/react/components/Map/Map.tsx

```diff
@@ -281,7 +333,81 @@ const Map = () => {
     dispatch,
   ]);
 
-  // Callbacks
+  const handleScrollEnd = useCallback(() => {
+    console.log("Handle Scroll End triggered with offset:", offset);
+
+    const hasMoreSessions = listSessions.length < fetchableMobileSessionsCount;
+
+    if (hasMoreSessions) {
+      const newOffset = offset + listSessions.length;
+      console.log("New offset to be set:", newOffset);
+      updateOffset(newOffset);
+
+      const updatedFilters = {
+        ...JSON.parse(filters),
+        offset: newOffset,
+      };
+
+      dispatch(
+        fetchMobileSessions({
+          filters: JSON.stringify(updatedFilters),
+          isAdditional: true,
+        })
+      )
+        .unwrap()
+        .then((response) => {
+          const totalFetchedSessions =
+            listSessions.length + response.sessions.length;
+          console.log("Updating fetched sessions:", totalFetchedSessions);
+          updateFetchedSessions(totalFetchedSessions);
+        });
+    }
+  }, [
+    offset,
+    listSessions.length,
+    fetchableMobileSessionsCount,
+    limit,
+    updateOffset,
+    dispatch,
+    filters,
+  ]);
+  // const handleScrollEnd = useCallback(() => {
+  //   const hasMoreSessions = listSessions.length < fetchableMobileSessionsCount;
+
+  //   if (hasMoreSessions) {
+  //     const newOffset = offset + listSessions.length;
+  //     updateOffset(newOffset);
+  //     console.log("setting newOffset", newOffset);
+
+  //     const updatedFilters = {
+  //       ...JSON.parse(filters),
+  //       offset: newOffset,
+  //     };
+
+  //     dispatch(
+  //       fetchMobileSessions({
+  //         filters: JSON.stringify(updatedFilters),
+  //         isAdditional: true,
+  //       })
+  //     )
+  //       .unwrap()
+  //       .then((response) => {
+  //         const totalFetchedSessions =
+  //           listSessions.length + response.sessions.length;
+  //         updateFetchedSessions(totalFetchedSessions);
+  //       });
+  //   }
+  // }, [
+  //   offset,
+  //   listSessions.length,
+  //   fetchableMobileSessionsCount,
+  //   limit,
+  //   updateOffset,
+  //   dispatch,
+  //   filters,
+  //   updateFetchedSessions,
+  // ]);
```

Comment:
Do we need this commented code?

Reply:
nope, this is another leftover -> I thought I removed it during rebase cleanup, I'll remove it